### ISSUE TITLE
draw with NaN params causes fatal memory leak

### ISSUE THREAD

**Issue by eyaler** [scikit-image/scikit-image#7114]
### Description:

Following code raises RAM usages to 100% within a few seconds and hangs the (64GB) machine. I had the NaN's due to an invalid bounding box calculation.

### Way to reproduce:

```
import numpy as np
from skimage.draw import rectangle_perimeter
rectangle_perimeter((np.nan, np.nan), (np.nan, np.nan))

```

### Version information:

```Shell
3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Windows-10-10.0.22621-SP0 (I am actually on Windows 11)
scikit-image version: 0.21.0
numpy version: 1.24.3
```

**Comment by lagru** [comment]
Hmm, I can't reproduce the RAM bombing on my machine. Instead I get
```python
skimage/draw/draw.py:959: RuntimeWarning: invalid value encountered in cast
  top_left = np.round(top_left).astype(int)
skimage/draw/draw.py:960: RuntimeWarning: invalid value encountered in cast
  bottom_right = np.round(bottom_right).astype(int)
skimage/draw/draw.py:271: RuntimeWarning: invalid value encountered in cast
  r = np.round(r).astype(int)
skimage/draw/draw.py:272: RuntimeWarning: invalid value encountered in cast
  c = np.round(c).astype(int)
```
and this output
```python
(array([-9223372036854775808, -9223372036854775808, -9223372036854775808,
        -9223372036854775808]),
 array([-9223372036854775808, -9223372036854775808, -9223372036854775808,
        -9223372036854775808]))
```
I am a bit surprised because from a casual look there isn't much that might be platform specific to the involved code. Maybe 
https://github.com/scikit-image/scikit-image/blob/a45f2daf9fcac32e3e8d2c95ade807ac67351965/skimage/draw/draw.py#L959-L960

is behaving different for you? Could you check the output of
```python
np.round((np.nan, np.nan)).astype(int)
```
on your machine?

In any case, we should probably introduce a guard that checks for NaN, which in this case shouldn't have any noticeable performance implications.

**Comment by lagru** [comment]
Oh, and thanks for the report @eyaler!

**Comment by eyaler** [comment]
@lagru 
```
> np.round((np.nan, np.nan)).astype(int)
<ipython-input-3-ea42ef702220>:1: RuntimeWarning: invalid value encountered in cast
  np.round((np.nan, np.nan)).astype(int)
Out[3]: array([-2147483648, -2147483648])
```
then you have in https://github.com/scikit-image/scikit-image/blob/v0.21.0/skimage/draw/draw.py#L929
`top_left -= 1`

so let do:
`a = np.round((np.nan, np.nan)).astype(int); a -= 1; print(a)`
and we get:
`array([2147483647, 2147483647])`

that's one big rectangle!

**Comment by lagru** [comment]
Yes, this doesn't seem to happen on my hardware because `np.round((np.nan, np.nan)).astype(int)` returns `array([-9223372036854775808, -9223372036854775808])` for me. That number is negative and get's treated differently. 

Are you interested in making a PR that adds a check for NaN's to address this issue? A simple `if np.isnan(...): raise ...` + a test should suffice. :D

**Comment by lagru** [comment]
Re my comment above, I actually forgot that I already made a PR for this https://github.com/scikit-image/scikit-image/pull/7115. :see_no_evil: :smile:
